access-token-management/src/AccessTokenManagement.OpenIdConnect/Implementation/OpenIdConnectClientAccessTokenRetriever.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Duende Software. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using Duende.AccessTokenManagement.OTel;
+using Microsoft.AspNetCore.Http;
+
+namespace Duende.AccessTokenManagement.OpenIdConnect.Implementation;
+
+/// <summary>
+/// TokenRetriever that implements the logic on behalf of <see cref="AccessTokenHandler{TTokenRetriever,TToken}"/> to
+/// retrieve a ClientCredentials Token for an OpenID Connect flow.
+/// </summary>
+/// <param name="httpContextAccessor"></param>
+/// <param name="parameters"></param>
+internal class OpenIdConnectClientAccessTokenRetriever(
```

Comment:
Consider renaming this, because we're obtaining a User access token, not a client (credentials) token.

Reply:
actually, we are retrieving a client credentials token